app/components/emoji_keyboard/emoji_keyboard.js

```diff
@@ -0,0 +1,248 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+    View,
+    Text,
+    StyleSheet,
+    FlatList,
+    TouchableOpacity
+} from 'react-native';
+import Icon from 'react-native-vector-icons/FontAwesome';
+import {KeyboardRegistry} from 'react-native-keyboard-input';
+
+import {store} from 'app/mattermost';
+import Emoji from 'app/components/emoji';
+import {
+    CategoryNames,
+    EmojiIndicesByCategory,
+    Emojis,
+    CategoryIconsFontAwesome
+} from 'app/utils/emojis';
+
+const EMPTY_STRING = '';
+const NUM_CATEGORIES = CategoryNames.length;
+
+class EmojiKeyboard extends React.PureComponent {
+    static propTypes = {
+        containerBackgroundColor: PropTypes.string
+    }
+    static defaultProps = {
+        containerBackgroundColor: 'white'
+    }
+    static EMOJI_SIZE = 32;
+    static BASE_PADDING = 4;
+    static CUSTOM_KEYBOARD_NAME = 'EmojiKeyboard';
+    static emojiRange = [...(new Array(NUM_CATEGORIES)).keys()]
+    static emojiCategories = CategoryNames.map((category) => (
+        {
+            key: category,
+            data: EmojiIndicesByCategory.get(category).map(
+                (index) => Emojis[index].aliases[0])
+        }
+    ));
+
+    static getCategoryNameByIndex = (index) => {
+        const nonNegativeIndex = Math.max(0, index);
+        const limited = Math.min(NUM_CATEGORIES - 1, nonNegativeIndex);
+        const category = EmojiKeyboard.emojiCategories[limited];
+        if (category && category.key) {
+            return category.key;
+        }
+        return EMPTY_STRING;
+    }
+
+    static getCategoryIconNameByIndex = (index) => {
+        return CategoryIconsFontAwesome[index];
+    }
+
+    state = {
```

Comment:
I think it would be best to have this at the top of the PureComponent near the `propType` definitions just for a better overview.